src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -512,36 +576,41 @@ public StorageTree Create(Address address, IScopedTrieStore trieStore, Hash256 s
             => new(trieStore, storageRoot, logManager);
     }
 
-    private sealed class DefaultableDictionary<TValue>(TValue defaultValue)
+    private sealed class DefaultableDictionary()
     {
         private bool _missingAreDefault;
-        private readonly Dictionary<UInt256, TValue> _dictionary = new(Comparer.Instance);
+        private readonly Dictionary<UInt256, ChangeTrace> _dictionary = new(Comparer.Instance);
 
         public void ClearAndSetMissingAsDefault()
         {
             _missingAreDefault = true;
             _dictionary.Clear();
         }
 
-        public ref TValue? GetValueRefOrAddDefault(UInt256 storageCellIndex, out bool exists)
+        public ref ChangeTrace GetValueRefOrAddDefault(UInt256 storageCellIndex, out bool exists)
         {
-            ref TValue value = ref CollectionsMarshal.GetValueRefOrAddDefault(_dictionary, storageCellIndex, out exists);
+            ref ChangeTrace value = ref CollectionsMarshal.GetValueRefOrAddDefault(_dictionary, storageCellIndex, out exists);
             if (!exists && _missingAreDefault)
             {
                 // Where we know the rest of the tree is empty
                 // we can say the value was found but is default
                 // rather than having to check the database
-                value = defaultValue;
+                value = ChangeTrace.ZeroBytes;
                 exists = true;
```

Comment:
Why then return `exists` in any case? I do understand that in `CollectionsMarshal` it's needed to create the default but here, we always set everything and exists is `true`. I'd remove it even though it won't be aligned with the `CollectionsMarshal`

Reply:
Only adds on `_missingAreDefault` which is after selfdestruct or creation of a new storage trie at vanilla address (that we know is empty)